site/sfguides/src/native-app-chairlift/native-app-chairlift.md

```diff
@@ -0,0 +1,641 @@
+
+
+<!-- ------------------------ -->
+## Set up account roles
+Duration: 3
+
+Let's start building the app. You'll first need to configure certain roles and permissions within your Snowflake account. This will allow you to view the app as an app admin (for configuring the application after installation and/or dismissing sensor warnings), or as an app viewer (perhaps someone in charge of equipment maintenance at the resort keeps an eye on the condition of chairlifts).
+
+To create these roles and permissions, run the scripts below. You'll only need to execute these scripts once.
+
+**Execute prepare/provider-role.sql**
+
+Open a SQL worksheet in Snowsight and execute the following script:
+
+```sql
+-- create provider role
+create role if not exists chairlift_provider;
+grant role chairlift_provider to role accountadmin;
+grant create application package on account to role chairlift_provider;
+grant create database on account to role chairlift_provider;
+
+-- ensure a warehouse is usable by provider
+create warehouse if not exists chairlift_wh;
+grant usage on warehouse chairlift_wh to role chairlift_provider;
+```
+
+**Execute prepare/consumer-roles.sql**
+
+Open a SQL worksheet in Snowsight and execute the following script:
+
+```sql
+-- create consumer role
+create role if not exists chairlift_admin;
+create role if not exists chairlift_viewer;
+grant role chairlift_admin to role accountadmin;
+grant role chairlift_viewer to role accountadmin;
+grant create database on account to role chairlift_admin;
+grant create application on account to role chairlift_admin;
+grant execute task, execute managed task on account to role chairlift_admin with grant option;
+grant role chairlift_viewer to role chairlift_admin;
+
+-- ensure a warehouse is usable by consumer
+grant usage on warehouse chairlift_wh to role chairlift_admin;
+grant usage on warehouse chairlift_wh to role chairlift_viewer;
+```
+
+
+<!-- ------------------------ -->
+## Prepare objects in account
+Duration: 4
+
+Next, you'll run some scripts to setup some databases, schemas, and tables needed by the app.
+
+The scripts will do a couple of things:
+
+* The provider scripts define roles, warehouses, and data types for the data emitted by the sensors (i.e., brake temperature, motor RPMs, etc.). This is necessary so that the app can function as intended in the consumer's account. From this perspective, the provider is the chairlift manufacturer, and the app will run against raw sensor data in (and owned by) a consumer's account. Running the app against this data will provide the consumer with analytics and insights on the chairlifts they own, granted that the consumer installs the app and grants the proper privileges to the app.
```

Comment:
Could mention that this data is shared with the consumer through the app instance.